documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -1837,3 +1838,23 @@ or
 ```
 
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4004" ></a>R4004 OperationIdRequired
+
+**Category** : ARM Warning
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** : Missing operationId. path:'${operation path}' operation:'${operation}'.
+
+**Description** : Each operation must has a unique operationId.
+
+**CreatedAt** : 2020-2-18
+
+**LastModifiedAt** : 2020-2-18
+
+**Why this rule is important**: Per [creating-swagger](creating-swagger.md#Paths),The operationId is used to determine the generated method name.
```

Comment:
Not quite understandable. The method name is built with operationId?  wonder if you want to express "the operationId is unique identification of the method" ?